ts-apps/decision-logic-visualizer/src/lib/data/viz-expr-to-lir.ts

```diff
@@ -26,38 +30,116 @@ export const VizDeclLirSource: LirSource<IRDecl, DeclLirNode> = {
       nodeInfo,
       decl.name,
       decl.params,
-      ExprLirSource.toLir(nodeInfo, decl.body)
+      LadderGraphLirSource.toLir(nodeInfo, decl.body)
     )
   },
 }
 
-export const ExprLirSource: LirSource<IRExpr, ExprLirNode> = {
-  toLir(nodeInfo: LirNodeInfo, expr: IRExpr): ExprLirNode {
-    return match(expr)
-      .with(
-        { $type: 'BoolVar' },
-        (originalVar) => new BoolVarLirNode(nodeInfo, originalVar)
-      )
-      .with(
-        { $type: 'Not' },
-        (n) => new NotLirNode(nodeInfo, ExprLirSource.toLir(nodeInfo, n.negand))
+/** Converts IRExpr to a Dag<LirId> (and sets the final dag in the LirContext)
+*
+* Preconditions / assumptions:
+* - To reduce visual clutter in the generated graph, the input ANDs/ORs should
+    have already been 'flattened'; i.e.,
+    nested exprs like (AND [a (AND [b])]) should have already been flattened to (AND [a b]).
+*
+*/
+export const LadderGraphLirSource: LirSource<IRExpr, LadderGraphLirNode> = {
+  toLir(nodeInfo: LirNodeInfo, expr: IRExpr): LadderGraphLirNode {
+    const overallSource = vertex(
+      new SourceLirNode(nodeInfo).getId()
```

Comment:
It might be a little bit more direct, if `LirId` had a static `getFreshId()`  or something like that, than you could avoid the indirection.
Just an idea, though.

Reply:
Ah what's happening here isn't that we are making a new id, but rather that we are making a new LirNode and then getting its id.